src/cls/tabular/fbwriter_skyroot.cc

```diff
@@ -0,0 +1,784 @@
+    // read data from file into general structure
+    // --------------------------------------------- //
+    filedata_t filedata ;
+    // initialize struct with empty vects
+    for( int i = 0; i < schema_datatypes.size(); i++ ) {
+      std::string this_dt = schema_datatypes[i] ;
+      if( this_dt == "int" ) {
+        std::string key = "att" + std::to_string( i ) + "-int" ;
+        // ---- >>>>
+        std::vector< std::string > an_index_pair ;
+        an_index_pair.push_back( key ) ;
+        an_index_pair.push_back( std::to_string( filedata.int_vect_cnt ) ) ;
+        filedata.indexer.push_back( an_index_pair ) ;
+        // <<<< ----
+        std::vector< uint64_t > an_empty_vect ;
+        filedata.listof_int_vect_raw.push_back( an_empty_vect ) ;
+        filedata.int_vect_cnt++ ;
+      }
+      else if( this_dt == "float" ) {
+        std::string key = "att" + std::to_string( i ) + "-float" ;
+        // ---- >>>>
+        std::vector< std::string > an_index_pair ;
+        an_index_pair.push_back( key ) ;
+        an_index_pair.push_back( std::to_string( filedata.float_vect_cnt ) ) ;
+        filedata.indexer.push_back( an_index_pair ) ;
+        // <<<< ----
+        std::vector< float > an_empty_vect ;
+        filedata.listof_float_vect_raw.push_back( an_empty_vect ) ;
+        filedata.float_vect_cnt++ ;
+      }
+      else if( this_dt == "string" ) {
+        std::string key = "att" + std::to_string( i ) + "-string" ;
+        // ---- >>>>
+        std::vector< std::string > an_index_pair ;
+        an_index_pair.push_back( key ) ;
+        an_index_pair.push_back( std::to_string( filedata.string_vect_cnt ) ) ;
+        filedata.indexer.push_back( an_index_pair ) ;
+        // <<<< ----
+        std::vector< std::string > an_empty_vect_strs ;
+        std::vector< flatbuffers::Offset<flatbuffers::String> > an_empty_vect ;
+        filedata.listof_string_vect_raw_strs.push_back( an_empty_vect_strs ) ;
+        filedata.listof_string_vect_raw.push_back( an_empty_vect ) ;
+        filedata.string_vect_cnt++ ;
+      }
+      else {
+        std::cout << ">>2 unrecognized schema_datatype '" << this_dt << "'" << std::endl ;
+        exit( 1 ) ;
+      }
+    } //for loop
+
+    // read csv (pipe delimited) from file
+    std::ifstream infile( inputs.filename ) ;
+    std::string line ;
+    int itnum = 0 ; //need this for some reason?
+    while ( infile.good() && itnum < nrows ) {
+      getline ( infile, line ) ;
+      std::vector< std::string > data_vect = parse_psv_str( line ) ;
+      for( int i = 0; i < data_vect.size(); i++ ) {
+        int attnum = i ;
+        std::string atttype = schema_datatypes[i] ;
+        std::string attkey  = "att" + std::to_string( attnum ) + "-" + atttype ;
+        uint64_t att_vect_id = filedata.get_index( attkey ) ;
+        std::string datum = data_vect[i] ;
+        std::cout << attkey << "," << att_vect_id <<  "," << datum << ";" ;
+
+        // int
+        if( atttype == "int" ) {
```

Comment:
is is possible to use col_info.type here? not sure, thanks.
https://github.com/uccross/skyhook-ceph/blob/skyhook-luminous/src/cls/tabular/cls_tabular_utils.h#L436